api-server/index.js

```diff
@@ -37,7 +37,24 @@ app.use(express.json())
 
 app.post('/project', async (req, res) => {
     const { gitURL, slug } = req.body
-    const projectSlug = slug ? slug : generateSlug()
+
+    // Regular expression to validate GitHub URL format
+    const githubUrlRegex = /^https?:\/\/github\.com\/[^\/]+\/[^\/]+$/;
+
+    // Regular expression to validate slug format (no spaces, alphanumeric characters)
+    const slugRegex = /^[a-zA-Z0-9-]+$/;
+
+    // Validate gitURL format
+    if (!urlRegex.test(gitURL)) {
+        return res.status(400).json({ status:"error", error: 'Please Enter Valid Git URL' });
+    }
+
+    // Validate slug format
+    if (slug && (!slugRegex.test(slug) || slug.includes(' '))) {
+        return res.status(400).json({ status:"error", error: 'Please Enter Valid Slug Format' });
+    }
+
+    const projectSlug = slug;
```

Comment:
If there is no `slug` provided by the user, we should generate one.

Reply:
So then I will have to remove the validation of slug from the request and generate the slug if it does not exist right?